recipe/emailpassword/override_test.go

```diff
@@ -0,0 +1,392 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package emailpassword
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/supertokens/supertokens-golang/recipe/emailpassword/epmodels"
+	"github.com/supertokens/supertokens-golang/recipe/session"
+	"github.com/supertokens/supertokens-golang/supertokens"
+	"github.com/supertokens/supertokens-golang/test/unittesting"
+)
+
+func TestOverridingFunctionCalls(t *testing.T) {
+	var user *epmodels.User
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			APIDomain:     "api.supertokens.io",
+			AppName:       "SuperTokens",
+			WebsiteDomain: "supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			Init(&epmodels.TypeInput{
+				Override: &epmodels.OverrideStruct{
+					Functions: func(originalImplementation epmodels.RecipeInterface) epmodels.RecipeInterface {
+						originalSignup := *originalImplementation.SignUp
+						originalSignIn := *originalImplementation.SignIn
+						originalGetUserById := *originalImplementation.GetUserByID
+						*originalImplementation.SignUp = func(email, password string) (epmodels.SignUpResponse, error) {
+							res, err := originalSignup(email, password)
+							if err != nil {
+								log.Fatal(err.Error())
+							}
+							user = &res.OK.User
+							return res, nil
+						}
+						*originalImplementation.SignIn = func(email, password string) (epmodels.SignInResponse, error) {
+							res, err := originalSignIn(email, password)
+							if err != nil {
+								log.Fatal(err.Error())
+							}
+							user = &res.OK.User
+							return res, nil
+						}
+						*originalImplementation.GetUserByID = func(userID string) (*epmodels.User, error) {
+							res, err := originalGetUserById(userID)
+							if err != nil {
+								log.Fatal(err.Error())
+							}
+							user = res
+							return res, nil
+						}
+						return originalImplementation
+					},
+				},
+			}),
+			session.Init(nil),
+		},
+	}
+
+	BeforeEach()
+	unittesting.StartUpST("localhost", "8080")
+	defer AfterEach()
+	err := supertokens.Init(configValue)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	mux := http.NewServeMux()
+	mux.HandleFunc("/user", func(rw http.ResponseWriter, r *http.Request) {
+		userId := r.URL.Query().Get("userId")
+		fetchedUser, err := GetUserByID(userId)
+		if err != nil {
+			t.Error(err.Error())
+		}
+		jsonResp, err := json.Marshal(fetchedUser)
+		if err != nil {
+			t.Errorf("Error happened in JSON marshal. Err: %s", err)
```

Comment:
Is using Errorf intentional or a mistake? if it's a mistake, please be sure to remove it from all tests, else resolve this comment